src/xinterpreter.cpp

```diff
@@ -116,6 +120,10 @@ namespace xpyt
             return kernel_res;
         }
 
+        py::module input_transformers = py::module::import("IPython.core.inputtransformer2");
+        py::object transformer_manager = input_transformers.attr("TransformerManager")();
+        py::str code_copy = transformer_manager.attr("transform_cell")(code);
```

Comment:
How hard do you think it would be to re-implement this in xeus-python?

Would it make more sense to implement it in C++ or should we extract the inputtransformer in a separate package to get rid of the IPython dependency?

Reply:
we need to investigate, but the parsing logic is rather nontrivial. For version 0, I would simply fail silently if IPython is not installed (and so the magic would not be supported without IPython installed).

Reply:
separating into a separate package is also an option, but I would not do that without discussing with IPython maintainers first (to avoid maintaining an independent code).

Reply:
> I would simply fail silently if IPython is not installed

Does it kill the kernel? It might be good if we show a warning saying that IPython must be installed in order to use magics.

Reply:
my idea is to try importing and if it fails, we will not parse the magics (and therefore no magic support). Of course, we might print a warning to the stdout or elsewhere.